src/ray/object_manager/plasma/common.h

```diff
@@ -45,23 +45,38 @@ enum class ObjectState : int {
   PLASMA_SEALED = 2,
 };
 
+// Represents a chunk of allocated memory.
+struct Allocation {
+  /// Pointer to the allocated memory.
+  void *address;
+  /// Num bytes of the allocated memory.
+  int64_t size;
```

Comment:
Better to use `size_t`.

Reply:
That's a great point. I'm thinking have a holistic fix of the types in plasma store once we finished 1st stage of refactoring; currently we'd stick with the existing behaviors to avoid unexpected suprises.

Reply:
> That's a great point. I'm thinking have a holistic fix of the types in plasma store once we finished 1st stage of refactoring; currently we'd stick with the existing behaviors to avoid unexpected suprises.

Awesome!

Reply:
I remember plasma store uses int64_t because some APIs in Arrow (the original plasma repo) uses -1 a lot for failures or invalid inputs. Maybe not change it so fast unless you entirely sure there won't be negative values. (alternatives like ssize_t are also options)